### PR TITLE
[Test] Add singleshot case test

### PR DESCRIPTION
**Pull request by zhoonit**
## Commits to be reviewed in this PR

<details><summary>[Filter] Add dimension inference mechanism</summary><br />

This patch add dimension inference mechanism as requested.
Please refer to nnstreamer/api#105 for details

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

</details>

<details><summary>[Test] Add singleshot case test</summary><br />

Add singleshot test case when ml_inference get null input information

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

</details>

### CODE REVIEW

File: test/nnstreamer/test_nnstreamer_single.cpp [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+#if false
+  status = singleshot_case(nullptr, out_info);
+  EXPECT_EQ(status, 0) << "case: in_info null, out_info given";
+
+  status = singleshot_case(in_info, nullptr);
+  EXPECT_EQ(status, 0) << "case: in_info given, out_info null";
+
+  status = singleshot_case(nullptr, nullptr);
+  EXPECT_EQ(status, 0) << "case: in_info null, out_info null";
+#endif
```
Comment (zhoonit) [inline test/nnstreamer/test_nnstreamer_single.cpp:142]: This is the gist that the PR tries to resolve.

File: nnstreamer/tensor_filter/tensor_filter_nntrainer.cc [hunk 2]
```diff
@@ -210,8 +218,22 @@ static int nntrainer_setInputDim(const GstTensorFilterProperties *prop,
     static_cast<NNTrainerInference *>(*private_data);
   g_return_val_if_fail(prop && nntrainer && in_info && out_info, -EINVAL);
 
-  auto num_input = in_info->num_tensors;
-  g_return_val_if_fail(num_input != 0, -EINVAL);
+  auto model_inputs = nntrainer->getInputDimension();
```
Comment (jijoongmoon) [inline nnstreamer/tensor_filter/tensor_filter_nntrainer.cc:221]: Good point. when the setInputDim is called, we need to set the dim from nntrainer information.

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nnstreamer/tensor_filter/tensor_filter_nntrainer.cc [hunk 3]
```diff
@@ -210,8 +218,22 @@ static int nntrainer_setInputDim(const GstTensorFilterProperties *prop,
...
+
+  /// if num_input is zero, we set a default input shape with batch of 1.
+  if (in_info->num_tensors == 0) {
```
Comment (kparichay) [inline nnstreamer/tensor_filter/tensor_filter_nntrainer.cc:224]: `in_info->num_tensors` indicates the number of inputs to be given to the model, and not the batch size.
Refer https://github.com/nnstreamer/nnstreamer/blob/c8d76bb71bd53dc16d4d48512a654da656aa3db0/ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc#L663.

File: nnstreamer/tensor_filter/tensor_filter_nntrainer.cc [hunk 4]
```diff
@@ -210,8 +218,22 @@ static int nntrainer_setInputDim(const GstTensorFilterProperties *prop,
...
+    ml_logi(
+      "num input is zero, regard it as a default dimension with batch of 1");
+    auto mutable_in_info = const_cast<GstTensorsInfo *>(in_info);
+    mutable_in_info->num_tensors = model_inputs.size();
+    for (unsigned int i = 0u; i < model_inputs.size(); ++i) {
+      auto default_dim = model_inputs[i];
+      default_dim.batch(1);
```
Comment (kparichay) [inline nnstreamer/tensor_filter/tensor_filter_nntrainer.cc:231]: this will override the batchsize set in ini? I wonder by batchsize has to be reset and not directly take what nntrainer provides. Note that nntrainer defaults batchsize to 1 if not specified.

Comment (zhoonit) [reply nnstreamer/tensor_filter/tensor_filter_nntrainer.cc:231]: consider the scenario where model is trained with batchsize of n and saved for inference.

It will be sensible that for inference default batchsize is 1, even though batchsize is specified.

Comment (kparichay) [reply nnstreamer/tensor_filter/tensor_filter_nntrainer.cc:231]: Then let's add in ini documentation that the batch size in ini will be ignored in inference.

**Review by kparichay — APPROVED** [review]
LGTM

File: nnstreamer/tensor_filter/tensor_filter_nntrainer.cc (hunk 4 above)
Comment (zhoonit) [reply nnstreamer/tensor_filter/tensor_filter_nntrainer.cc:231]: agreed :)
